datagen/src/main/spark320/scala/org/apache/spark/sql/tests/datagen/datagen/DataGenExprBase.scala

```diff
@@ -36,7 +36,6 @@
 {"spark": "341db"}
 {"spark": "342"}
 {"spark": "350"}
-{"spark": "351"}
```

Comment:
Why does this PR remove 351 when it says it's adding shims?

Reply:
it's actually adding the 334 and 342 shim versions to the pom file for scala 2.13.  

351 context is here: https://github.com/NVIDIA/spark-rapids/pull/10465#issuecomment-1965533960

Reply:
We should _not_ be removing 351 in the source code this late.  Do we want to not build 351 support in the 24.02 jar?  Fine.  But this much source churn this late is not good.  Modify the pom to not ask for a 351 build rather than fully ripping out the source code.

Reply:
And this will cause a merge nightmare for 24.04 if done this way.